exercises/08-execute-javascript.md

```diff
@@ -85,17 +89,49 @@ We notice that the "Important number" element has an attribute called `value` an
 our `wanted_value` argument. So, full execution would be
 `document.evaluate(...).singleNodeValue.value = ${wanted_value};`.
 
-As we now have our `Execute Javascript` implementation ready, we can change the logic inside
-`Fill All Form Fields` keyword to change the important number with JavaScript.
-
 - Write `Execute Javascript` implementation using either the XPath or CSS selector to change the "Important
 number" value to `wanted_value`.
-- Add `${TRUE}` as a final argument when you call `Change Important Number` inside `Fill All Form Fields`.
 
 > `Execute Javascript` can return values normally, but in order to get values
 > you need to explicitly tell the JavaScript command to `return` even though
 > the command would normally return a value when running in the browser console,
 > e.g. `${element}=       Execute Javascript    return document.getElementById("myId");`
+</details> <!-- SeleniumLibrary -->
+
+<details>
+  <summary>Browser</summary>
+
+In contrast to SeleniumLibrary's implementation, Browser's keyword to execute JavaScript takes two arguments. The second argument is
+a reference to an element in the web page, and the first one is the JS function,
+that will be sent to that referenced element. Locator is not the same as a reference, so we will need to call another keyword to create a reference first.
+
+The operation on the slider is thus not a one-liner, so let's make a
+separate keyword for it, which can be then called by `Run Keyword If`.
+The new keyword should contain the steps to get a reference for the slider, and change its value with a simple JS function.
+
+- Add a keyword `Change slider value with JS` that will change the slider's value to `wanted_value`.
+
+Reference to an element is returned by `Get Element` keyword, which takes a
+locator as an argument. Store it in a local variable. It can be used like a
+locator in other keywords, even without the need to specify the iframe.
+With this direct reference to the slider, JS function needed is a simple change of the `value` property.
+Example function syntax: `(element) => element.property = "new value"`.
+
+- Get and store slider's reference, and use it to assign `wanted_value` to slider's `value` property.
+
+Keyword should also check that the value stored by the slider indeed changed.
+This can be done by confirming the value before the change is `0`, and after is
+equal to the `wanted value`. Browser's keywords have built-in checking, so
+there's no need to store the numbers and compare them as integers. The value is stored inside slider's properly, so you can use the `Get Property` keyword.
+
+- Implement checks with `Get Property` to ensure the value has been changed.
```

Comment:
Keyword validation is explained later on in the exercise file, but it's used here for the first time, so it should be explained here:

```suggestion
Browser library has builtin waiting and no separate validation keywords. Instead, you can use `Get Text`
to validate the text automatically. Python validations such as `==`, `!=`, `contains`, and `not contains`
are available for validation. The syntax is Python-esque:

```robot
Get Property    locator    ==    some value
```

- Implement checks with `Get Property` to ensure the value has been changed.
```

Reply:
Code blocked moved here.